bcda/auth/client/oktaclient.go

```diff
@@ -31,94 +42,213 @@ func init() {
 	} else {
 		logger.Info("No Okta log location provided; using default stderr")
 	}
+	// there is no possibility of recovery if we put the call to config() here
+	// because init() is called once when main is being started
+}
 
+// separate from init for testing
+func config() error {
+	// required env vars
+	oktaBaseUrl = os.Getenv("OKTA_CLIENT_ORGURL")
+	oktaServerID = os.Getenv("OKTA_OAUTH_SERVER_ID")
 	oktaToken := os.Getenv("OKTA_CLIENT_TOKEN")
-	if oktaToken == "" {
-		logger.Fatal("No Okta token found; please set OKTA_CLIENT_TOKEN")
+
+	// report missing env vars
+	at := oktaToken
+	if at != "" {
+		at = "[Redacted]"
 	}
 
-	oktaURL := os.Getenv("OKTA_CLIENT_ORGURL")
-	if oktaURL == "" {
-		logger.Fatal("No Okta URL found; please set OKTA_CLIENT_ORGURL")
+	if oktaBaseUrl == "" || oktaServerID == "" || oktaToken == "" {
+		return fmt.Errorf(fmt.Sprintf("missing env vars: OKTA_CLIENT_ORGURL=%s, OKTA_OAUTH_SERVER_ID=%s, OKTA_CLIENT_TOKEN=%s", oktaBaseUrl, oktaServerID, at))
 	}
+
+	// manufactured from env var
+	oktaAuthString = fmt.Sprintf("SSWS %s", oktaToken)
+
+	return nil
 }
 
-func HealthCheck() error {
-	reqId := uuid.NewRandom()
-	client := NewOktaClient()
+type OktaClient struct {}
 
-	// The email in OKTA_EMAIL should represent a test user present in the Okta sandbox environment
-	oktaTestUser, success := os.LookupEnv("OKTA_EMAIL")
-	if !success {
-		logRequest(reqId).Warn("Unable to perform Okta HealthCheck.  Please set OKTA_EMAIL to match a test user account.")
-		return errors.New("cannot perform Okta health check without OKTA_EMAIL configured ")
+// Returns an OktaClient. An OktaClient is always created, whether or not it is currently able to converse with Okta.
+func NewOktaClient() *OktaClient {
+	var err error
+	once.Do(func() {
```

Comment:
We keep a list of our okta authorization server's public keys so that we can decode / validate tokens. We only need to get the keys the first time an OktaClient value is created.